### ISSUE TITLE
File changes that happen while a watch task is running get ignored

### ISSUE THREAD

**Issue by russelldavis** [gruntjs/grunt-contrib-watch#259]
This is an expanded version of issue #33. That issue was closed, but the problem still exists since the commit that fixed it got reverted. I have some more details that should be helpful.

Start with this Gruntfile.js (slightly modified from the one is issue 33):

```
module.exports = function(grunt) {
  grunt.initConfig({
    watch: {
      gruntfile: {
        files: 'Gruntfile.js',
        tasks: ['longtask']
      },
      shorttask: {
        files: 'shorttask',
        tasks: ['shorttask']
      }
    }
  });

  grunt.loadNpmTasks('grunt-contrib-watch');

  grunt.registerTask('shorttask', function(){
    console.log('shorttask');
  });

  grunt.registerTask('longtask', function(){
    var originalGrunt = grunt.file.read('Gruntfile.js');
    var done = this.async();
    setTimeout(function(){
      grunt.file.write('Gruntfile.bak', originalGrunt);
      done();
    }, 10000);
  });

  grunt.registerTask('default', 'watch');
};
```

Create a file called `shorttask` in the same directory. Now, run `grunt watch`, then `touch Gruntfile.js`. Then, while 'longtask' is still running, run `touch Gruntfile.js` again. You'll notice that 'longtask' does _not_ run again after it finishes. This was the original problem reported in issue 33.

However, you'll notice if you change the last step in the paragraph above to `touch shorttask`, then 'shorttask' _will_ run after 'longtask' finishes. So, the problem above only applies to modifying files from the same watch group as the task that is running. I believe this is due to [this line](https://github.com/gruntjs/grunt-contrib-watch/blob/master/tasks/watch.js#L160) in watch.js.

But wait, there's more. If you repeat that, but first add `options: {nospawn: true}` to the watch config, then 'shorttask' will _not_ run after 'longtask' finishes. When `nospawn` is set, the problem applies regardless of which watch group is modified.

To summarize:
1) Regardless of `nospawn`, modifications to a file from a watch group whose tasks are currently running get ignored.
2) When `nospawn` is true, modifications to a file from any watch group get ignored when any watch group's tasks are running.
